river/View.zig

```diff
@@ -492,18 +493,35 @@ pub fn map(view: *Self) !void {
         view.pending.ssd = ssd;
     }
 
-    if (server.input_manager.defaultSeat().focused_output) |output| {
+    const output: ?*Output = blk: {
+        if (server.config.output_rules.match(view)) |output_name| {
+            var it = server.root.active_outputs.iterator(.forward);
+            while (it.next()) |output| {
+                // Checks if output_name matches with either the connector or
+                // name or full output description
+                if (mem.eql(u8, mem.span(output.wlr_output.name), output_name) or
+                    (output.wlr_output.description != null and
+                    mem.eql(u8, mem.span(output.wlr_output.description.?), output_name)))
+                {
+                    break :blk output;
+                }
+            }
+        }
+        break :blk server.input_manager.defaultSeat().focused_output;
+    };
+
+    if (output != null) {
```

Comment:
I think this could be cleaned up a bit.

The part checking for an output rule match is a bit verbose and the details of the search are not relevant to the rest of the View.map() function. I think it belongs in a different function, `Config.zig` would be a fine place for such a function.

Then, one could write something like
```zig
if (Config.outputRuleMatch(view) orelse focused_output) |output| {
...
}
```
In any case, the `.?` unwrapping in the block below is unnecessary and should be eliminated with an unwrapping `if (foo) |bar|` if statement instead of a `!= null` check.